2020/cmd/08/vm.go

```diff
@@ -63,6 +64,9 @@ func parseInstruction(s string) (instr Instruction, err error) {
 
 	switch matches[1] {
 	case "nop":
+		// QUESTION: this one is weird to me. It does not compile if I expect `*Instruction` in params.
+		// Why?
+		// NOTE: cannot use &(nopInstr literal) (value of type *nopInstr) as *Instruction value in assignment
 		instr = &nopInstr{v: num}
 		return
```

Comment:
Fot God's sake do not use bare returns, especially in long functions.
Use `return instr, nil`